src/harness/reference_models/geo/utils.py

```diff
@@ -442,19 +442,28 @@ def GetClosestCanadianBorderPoint(latitude, longitude, max_dist_km):
   max_delta = max_dist_km / one_degree * 1.1
   border_cap = zones.GetUsCanadaBorder().intersection(
       sgeo.Point(longitude, latitude).buffer(max_delta))
-  if not border_cap or 'LineString' not in border_cap.type:
+  if not border_cap:
     return None
+  points_dists = []
   # Find closest point
-  points_dists = [(vincenty.GeodesicDistanceBearing(latitude, longitude, point[1], point[0]),
-                   point)
-                  for point in zip(*border_cap.xy)]
+  if border_cap.type == 'LineString':
+    points_dists.extend(_distancesOfLineString(latitude, longitude, border_cap))
+  elif border_cap.type == 'MultiLineString':
+    for line in border_cap.geoms:
```

Comment:
Could be sufficient to do:
```for line in border_cap:```

All the MultiXXX thing are iteratating on the non Multi version